tests/test_hardhat.py

```diff
@@ -83,30 +83,31 @@ def test_multiple_hardhat_instances(network_api):
     provider_1.mine()
     provider_2.mine()
     provider_3.mine()
-    hash_1 = provider_1._web3.eth.get_block("latest").hash
-    hash_2 = provider_2._web3.eth.get_block("latest").hash
-    hash_3 = provider_3._web3.eth.get_block("latest").hash
+    hash_1 = provider_1.get_block("latest").hash
+    hash_2 = provider_2.get_block("latest").hash
+    hash_3 = provider_3.get_block("latest").hash
     assert hash_1 != hash_2 != hash_3
 
 
 def test_set_block_gas_limit(hardhat_provider):
-    gas_limit = hardhat_provider._web3.eth.get_block("latest").gasLimit
+    gas_limit = hardhat_provider.get_block("latest").gas_data.gas_limit
     assert hardhat_provider.set_block_gas_limit(gas_limit) is True
 
 
-def test_sleep(hardhat_provider):
+def test_set_timestamp(hardhat_provider):
     seconds = 5
-    time_1 = hardhat_provider.sleep(seconds)
-    time_2 = hardhat_provider.sleep(seconds)
-    assert time_2 - time_1 == seconds
+    time_1 = hardhat_provider.get_block("pending").timestamp
+    hardhat_provider.set_timestamp(seconds + time_1)
+    time_2 = hardhat_provider.get_block("pending").timestamp
+    actual = time_2 - time_1
+    assert abs(actual - seconds) <= 1
```

Comment:
I don't think we can use approx here, since these are integers are we are not checking precision. I tried to see if we can use it here but couldn't find a way, just a lot of false positives

I can do this though: `assert actual in (-1, 0, 1)`

Reply:
Please correct me if I am wrong, I can't figure out how to use approx here. I can only get that to work with floats, but itmestamps are ints and seconds for blocks

Reply:
ah okay. this might be a bit clearer though:
```suggestion
    assert -1 <= actual - seconds <= 1
```

Reply:
I think it can just be `assert -1 < actual < 1`! Which is not bad.